src/fsharp/MethodOverrides.fs

```diff
@@ -420,7 +459,116 @@ module DispatchSlotChecking =
                     | meth :: _ when meth.IsFinal -> errorR(Error(FSComp.SR.tcCannotOverrideSealedMethod((sprintf "%s::%s" (meth.ApparentEnclosingType.ToString()) (meth.LogicalName))), m))
                     | _ -> ()
 
+    /// Get default slot flags for the given method.
+    let inline GetDefaultDispatchSlotFlags (minfo: MethInfo) =
+        (if minfo.IsAbstract then RequiredSlotFlags.None else RequiredSlotFlags.Optional) |||
+        (if minfo.IsDefaultInterfaceMethod then RequiredSlotFlags.DefaultInterfaceImplementation else RequiredSlotFlags.None)
 
+    /// Get a collection of interface methods that are overrider (aka. override by) slots for the given method from the given types.
+    let GetInterfaceOverriderMethods (infoReader: InfoReader) m (minfo: MethInfo) tys =
+        let g = infoReader.g
+        let amap = infoReader.amap
+        let mty = minfo.ApparentEnclosingType
+
+        if minfo.IsInterfaceMethod then
+            tys
+            |> List.choose (fun ty ->
+                if isInterfaceTy g ty && not (typeEquiv g mty ty) && TypeFeasiblySubsumesType 0 g amap m mty CanCoerce ty then
+                    // We are only looking for IL overriders because F# does not support creating your own default interface methods.
+                    match TryFindILIntrinisicOverriderMethInfo infoReader m AccessibleFromSomewhere ty minfo with
+                    | Some minfo2 when not minfo2.IsNewSlot -> Some minfo2
+                    | _ -> None
+                else
+                    None)
+        else
+            []
+
+    /// Get a collection of interface methods that are the top most overrider methods in the hierarchy for the given method from the required type infos.
+    let GetTopMostHierarchicalInterfaceOverriderMethods (infoReader: InfoReader) m reqdTyInfos (minfo: MethInfo) =
+        let g = infoReader.g
+        let amap = infoReader.amap
+
+        if not minfo.IsInterfaceMethod then
+            failwith "method is not on an interface"
+
+        if not minfo.IsNewSlot then
+            failwith "method is not a new slot"
+        
+        let sortedOverriderMethods =
+            reqdTyInfos
+            |> List.map (fun (_, _, _, impliedTys) ->
+                GetInterfaceOverriderMethods infoReader m minfo impliedTys
+            )
+            |> List.reduce (@) // ok to use reduce here without checking as we will always have 1 item
+            // Sort the hierarchy to determine the best possible method.
+            |> List.sortWith (fun minfo1 minfo2 ->
+                if TypeFeasiblySubsumesType 0 g amap m minfo1.ApparentEnclosingType CanCoerce minfo2.ApparentEnclosingType then 1
+                else -1
+            )
+                 
+        match sortedOverriderMethods with
+        | [] ->
```

Comment:
Adding an explanatory comment to each case would be helpful

Reply:
Yea, for sure, it really isn't clear